### DISCUSSION TITLE
What does "First class support for Anvil and Hardhat" mean and why isn't Ganache supported?

### DISCUSSION THREAD

**Discussion by davidmurdoch** [wevm/viem#124] [Question] [answered]
What does "First class support for Anvil and Hardhat" mean and why isn't Ganache supported?

**Comment by jxom** [comment, marked as answer]
We have first class support for "test" RPC methods that Anvil & Hardhat provides.

[Learn more here](https://viem.sh/docs/clients/test.html).

I am not too familiar with Ganache, but if they also support methods like `evm_mine`, `evm_increaseTime`, etc, then we can easily integrate it in!

**Reply by davidmurdoch** [reply]
Ah. Ganache actually invented those methods and the `evm` namespace [in 2016](https://github.com/trufflesuite/ganache-cli-archive/commit/8d373d97700c618f0a68c253803be6d4f1153cee).

You can explore and play with Ganache's RPC methods in browser here: https://ganache.dev

EVM RPC docs are here: https://ganache.dev/#evm_addAccount

p.s., the blur feature on your docs to focus the user's attention are really great :fire: !

**Reply by jxom** [reply]
Sweet! That's amazing. 

Added it to the tracking [Roadmap](https://github.com/orgs/wagmi-dev/projects/3/views/1?pane=issue&itemId=22184144) – anticipate this will be a pretty trivial addition, will take a look soon!

Love Ganache UI btw.

**Reply by SonOfMosiah** [reply]
Would also love to see support for Tenderly under the same umbrella. Similar support for `evm_mine`, `evm_increaseBlocks`, etc. but also additional methods such as `tenderly_setStorageAt` (`hardhat_setStorageAt` equivalent).
